From c5639a33321280ebd866c0bce13cb24618faf9d8 Mon Sep 17 00:00:00 2001
From: Xufeng Zhang <xufeng.zhang@windriver.com>
Date: Tue, 28 Jun 2011 16:59:04 +0800
Subject: [PATCH 14/47] t23x: fix compile error

Extracted from vendor drop QorIQ-NONDPAA-SDK-V1-20110429_ltib.iso
linux-2.6.35-qoriq-t23x-fix-compile-error.patch and
linux-2.6.35-qoriq-t23x-fix-compile-fail.patch.

Signed-off-by: Li Yang <leoli@freescale.com>
Signed-off-by: Jin Qing <b24347@freescale.com>
Integrated-by: Xufeng Zhang <xufeng.zhang@windriver.com>
---
 drivers/crypto/t23x/t23xrm/t23xrmMemUtil.c  |    3 ++-
 drivers/crypto/t23x/t23xrm/t23xrmResponse.c |    4 ++--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/crypto/t23x/t23xrm/t23xrmMemUtil.c b/drivers/crypto/t23x/t23xrm/t23xrmMemUtil.c
index c62f1e3..79fdab5 100644
--- a/drivers/crypto/t23x/t23xrm/t23xrmMemUtil.c
+++ b/drivers/crypto/t23x/t23xrm/t23xrmMemUtil.c
@@ -6,7 +6,7 @@
  * interfaces
  *
  *
- * Copyright (c) 2007-2009 Freescale Semiconductor, Inc.
+ * Copyright (c) 2007-2010 Freescale Semiconductor, Inc.
  *
  * All rights reserved.
  *
@@ -62,6 +62,7 @@
 #include <asm/page.h>
 #include <asm/io.h>
 #include <asm/scatterlist.h>
+#include <linux/slab.h>
 
 #include "../common/xwcRMinterface.h"
 #include "t23xrmInternal.h"
diff --git a/drivers/crypto/t23x/t23xrm/t23xrmResponse.c b/drivers/crypto/t23x/t23xrm/t23xrmResponse.c
index 537649e..d7b3547 100644
--- a/drivers/crypto/t23x/t23xrm/t23xrmResponse.c
+++ b/drivers/crypto/t23x/t23xrm/t23xrmResponse.c
@@ -5,7 +5,7 @@
  * t23x resource manager response processing module. This module
  * handles post-request processing on behalf of an interface
  *
- * Copyright (c) 2007-2009 Freescale Semiconductor, Inc.
+ * Copyright (c) 2007-2010 Freescale Semiconductor, Inc.
  *
  * All rights reserved.
  *
@@ -342,7 +342,7 @@ void t23RMdeferredInterruptResponder(int inst)
                 burstsize--;
             }
         }
-        spin_unlock_irqrestore(thisDev->execQlock, irqflags);
+        spin_unlock_irqrestore(&thisDev->execQlock, irqflags);
 
     } while (thisDev->isrQlevel);
 }
-- 
1.7.0.4

